### ISSUE TITLE
#2140 seems to have broken apt on debian somehow

### ISSUE THREAD

**Issue by GodloveD** [apptainer/singularity#2176]
## Version of Singularity:

Current master.  But seems to have broken on  2c31aaaf166262b738cb350fe7fc390552845f4a.

### Expected behavior

One should be able to able to build a container with the following def file
```
Bootstrap: library
From: debian:9

%post
    apt-get -y update
```
### Actual behavior

Before 2c31aaaf166262b738cb350fe7fc390552845f4a, this worked just fine.  But after 2c31aaaf166262b738cb350fe7fc390552845f4a one gets the following results. 

```
$ sudo singularity build debian.sif debian.def 
WARNING: Authentication token file not found : Only pulls of public images will succeed
INFO:    Starting build...
 40.36 MiB / 40.36 MiB [=========================================] 100.00% 3.32 MiB/s 12s
INFO:    Running post scriptlet
+ apt-get -y update
Ign:1 http://cdn-fastly.deb.debian.org/debian stretch InRelease                    
Get:2 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]  
Get:4 http://security.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Get:3 http://cdn-fastly.deb.debian.org/debian stretch Release [118 kB]               
Get:5 http://cdn-fastly.deb.debian.org/debian stretch Release.gpg [2434 B]                  
Get:6 http://cdn-fastly.deb.debian.org/debian stretch-updates/main amd64 Packages [5148 B]
Get:7 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 Packages [7099 kB]        
Get:8 http://security.debian.org/debian-security stretch/updates/main amd64 Packages [447 kB]
Fetched 7856 kB in 4s (1582 kB/s)                               
Reading package lists... Done
E: Unable to change to (unreachable)/home/david/sandbox/ - chdir (2: No such file or directory)
FATAL:   post proc: exit status 100
FATAL:   While performing build: while running engine: exit status 255
```

### Steps to reproduce behavior

Debug output doesn't provide many clues.
```
[...snip...]
DEBUG   [U=0,P=86534]      CreateContainer()             Chdir into / to avoid errors
DEBUG   [U=0,P=86542]      set_parent_death_signal()                 Set parent death signal to 9
DEBUG   [U=0,P=86542]      prepare_scontainer_stage()                Entering in scontainer stage 2
DEBUG   [U=0,P=86542]      set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=0,P=86542]      startup()                     Execute scontainer stage 2
INFO    [U=0,P=86542]      StartProcess()                Running post scriptlet
+ apt-get -y update
Get:1 http://security.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Get:2 http://security.debian.org/debian-security stretch/updates/main amd64 Packages [447 kB]
Ign:3 http://cdn-fastly.deb.debian.org/debian stretch InRelease                                                                                                                         
Get:4 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:5 http://cdn-fastly.deb.debian.org/debian stretch Release [118 kB]
Get:6 http://cdn-fastly.deb.debian.org/debian stretch-updates/main amd64 Packages [5148 B]
Get:7 http://cdn-fastly.deb.debian.org/debian stretch Release.gpg [2434 B]
Get:8 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 Packages [7099 kB]
Fetched 7856 kB in 16s (482 kB/s)   
Reading package lists... Done
E: Unable to change to (unreachable)/home/david/sandbox/ - chdir (2: No such file or directory)
FATAL   [U=0,P=86542]      StartProcess()                post proc: exit status 100
DEBUG   [U=0,P=86534]      SMaster()                     Child exited with exit status 255
FATAL   [U=0,P=86520]      func6()                       While performing build: while running engine: exit status 255
```

**Comment by jmstover** [comment]
The exact commit that breaks this is: https://github.com/sylabs/singularity/pull/2140/commits/79e22a27a16f716d67bc01253f7d1bdedb30b9c1

```
$ /usr/local/singularity/3.0.0-pr2140-6/bin/singularity version                                
v3.0.0-beta.1-100-g79e22a2

$ sudo /usr/local/singularity/3.0.0-pr2140-6/bin/singularity build debian_test1.sif debian_test.def 
WARNING: Authentication token file not found : Only pulls of public images will succeed
INFO:    Starting build...
 40.36 MiB / 40.36 MiB [========================================================================================================================================] 100.00% 9.94 MiB/s 4s
INFO:    Running post scriptlet
+ apt-get update
Get:1 http://security.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Ign:2 http://cdn-fastly.deb.debian.org/debian stretch InRelease                
Get:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:4 http://cdn-fastly.deb.debian.org/debian stretch Release [118 kB]            
Get:5 http://security.debian.org/debian-security stretch/updates/main amd64 Packages [447 kB]
Get:6 http://cdn-fastly.deb.debian.org/debian stretch Release.gpg [2434 B]              
Get:7 http://cdn-fastly.deb.debian.org/debian stretch-updates/main amd64 Packages [5148 B]      
Get:8 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 Packages [7099 kB]         
Fetched 7856 kB in 1s (4882 kB/s)                             
Reading package lists... Done
E: Unable to change to (unreachable)/jason/tmp/sing/ - chdir (2: No such file or directory)
FATAL:   post proc: exit status 100
FATAL:   While performing build: while running engine: exit status 255
```

The commit prior (aa22c1b) worked fine.

**Comment by cclerget** [comment]
Fix included with #2158, added CLONE_FS flag during fork to propagate chroot/chdir to parent process running `StartProcess`

**Comment by GodloveD** [comment]
It doesn't seem that #2158 is fixing this issue from what I can tell.

**Comment by bauerm97** [comment]
Fixed by #2158
